File: Java/1685. Sum of Absolute Differences in a Sorted Array.java
class Solution {
    public int[] getSumAbsoluteDifferences(int[] nums) {
        int left_sum = 0, right_sum = 0, l = nums.length;
        for (int n: nums) {
            right_sum += n;
        }

        for (int i = 0; i < l; i++) {
            int temp = nums[i] * i - left_sum + right_sum - nums[i] * (l - i);
            left_sum += nums[i];
            right_sum -= nums[i];
            nums[i] = temp;  
        }

        return nums;
    }
}

File: Python3/1685. Sum of Absolute Differences in a Sorted Array.py
class Solution:
    def getSumAbsoluteDifferences(self, nums: List[int]) -> List[int]:
        left_sum = 0
        right_sum = sum(nums)
        size = len(nums)
        ans = [-1] * size

        for i, n in enumerate(nums):
            val = n * i - left_sum + right_sum - n * (size - i)
            ans[i] = val
            left_sum += n
            right_sum -= n

        return ans